### PR TITLE
Rc

### PR DESCRIPTION
**Pull request by franciscotln**
`test-helpers2.js` is the new file;
`test-helpers.js` is your file but wrapped in a module on the window, so window._Test. So instead of TestServ() you should call _Test.Serv() and that's your version of it.

I've added a underscore because your helpers are still being used in the Test.Factory examples so I didn't want to crash the tests now. It's just a TODO thing. Except for this part, I updated all tests (with and without helpers).

There are some comments in the "dummy example section", please read.
The ReadMe would have to be updated, of course.

### CODE REVIEW

File: examples/TestDummy/directive-with-dummy.js [hunk 1]
```diff
@@ -26,27 +26,22 @@ angular
...
-      $rootScope = _$rootScope_;
-      $scope = $rootScope;
-      element = angular.element('<some-directive></some-directive>');
-      $compile(element)($scope);
-      angular.element(document.body).append(element);
-      $scope.$apply();
-    });
+  var Test, element, $scope;
+
+  module.sharedInjector();
```
Comment (dakolech) [inline examples/TestDummy/directive-with-dummy.js]: Angular.mock.module.sharedInjector is available from version 1.5.1 (https://code.angularjs.org/1.5.1/docs/api/ngMock/function/angular.mock.module.sharedInjector). You should add somewhere comment with this information.

File: examples/TestDummy/filter-with-dummy.js [hunk 2]
```diff
@@ -1,52 +1,45 @@
...
-      $compile(element)($scope);
-      angular.element(document.body).append(element);
-      $scope.$apply();
-    });
+  var Test, element, $scope;
+
+  module.sharedInjector();
+
+  beforeAll(function() {
+    // FIXME: Do we need the Dummy.Filter? It's easier to pass an empty function(){};
```
Comment (dakolech) [inline examples/TestDummy/filter-with-dummy.js]: Dummy filter and directive are only aliases. It's easier to remember words than choose between function and object.

File: examples/TestElement/controller-with-test-element.js [hunk 3]
```diff
@@ -16,35 +18,28 @@ angular
...
-    spyOn(mockedSomeService, 'create').and.returnValue({
-      then: function(success, fail) {
-        successCallback = success;
-        failCallback = fail;
-      }
-    });
+  var Test, someController, successCallback, failCallback, mockedSomeService, mockedAlerts;
+
+  mockedSomeService = {
+      create: jasmine.createSpy('create').and.returnValue({
```
Comment (dakolech) [inline examples/TestElement/controller-with-test-element.js]: indentation, and I prefer creating a new spy before each test

File: examples/TestServ/with-test-serv.js [hunk 4]
```diff
@@ -1,9 +1,10 @@
-jasmine.pp = function(obj) {
-  return JSON.stringify(obj, undefined, 2);
-}
+'use strict';
+
+// jasmine.pp = function(obj) {
+//   return JSON.stringify(obj, undefined, 2);
+// }
```
Comment (dakolech) [inline examples/TestServ/with-test-serv.js]: you can remove this

File: test-helpers2.js [hunk 5]
```diff
@@ -0,0 +1,407 @@
...
+  }
+
+  function Serv(name) {
+    if (name) {
+      return core.$injector.get(name);
+    }
+  }
+
+  // This method allows you to add any number of methods as a list of string method names which return promises like so:
+  // myMockedService.addPromise('methodA', 'methodB', 'methodC');
```
Comment (dakolech) [inline test-helpers2.js]: If you are adding comments here, you should use JSDoc convention: http://usejsdoc.org/about-getting-started.html

**Comment by dakolech** [comment]
I'm not sure if using beforeAll instead of  beforeEach in every spec file is a good convention. These examples are very simple, so it's working, but in more complicated tests we should mix them.

And I will not merge it to the master if it isn't ready.

**Comment by HipsterZipster** [comment]
I agree we should definitely not use beforeAll instead of beforeEach.

**Pull request closed without merging**